server/ai_live_video.go

```diff
@@ -336,9 +335,17 @@ func startControlPublish(ctx context.Context, control *url.URL, params aiRequest
 		done <- true
 	}
 
+	if control, exists := params.node.LivePipelines[stream]; exists {
+		clog.Info(ctx, "Stopping existing control loop", "existing_request_id", control.RequestID)
+		control.StopControl()
```

Comment:
I wonder if it could cause deadlock or something, you call with mutex locked ant then it writes to the `done` channel? A simple workaround would be to increase the done channel buffer size

Reply:
Yeah it is not a very safe API, if StopControl is called more times than the channel capacity then we are blocked...

Improved the situation a bit in f8d4498d1ffab43ef3570af036141de0ce432355 but also changed it to close the control channel instead of the StopControl function [1]. This will make things bounce around a bit if the runner still has the control channel open but that should eventually make the gateway close the stream associated with the old request id, if it's still coming in.

[1] One day we will close the control channel inside the StopControl function!